crates/yulgen/src/mappers/functions.rs

```diff
@@ -128,23 +129,23 @@ fn revert(context: &mut Context, stmt: &Node<fe::FuncStmt>) -> yul::Statement {
                 .get_expression(error_expr)
                 .expect("missing expression");
 
-            if let Type::Struct(val) = &error_attributes.typ {
-                context.revert_errors.insert(val.clone());
+            if let Type::Struct(_struct) = &error_attributes.typ {
+                context.revert_errors.insert(_struct.clone());
```

Comment:
note how we can get rid of this

Reply:
@cburgdorf 

Instead of adding this information to the context and then building a list of Yul functions based on it, I think we should accumulate functions as operations are used. This should make things more manageable.

Basically, inside of the operation function, we would have a line like `context.add_deps(operation_deps)`, where `operation_deps` is a list functions needed in the Yul runtime for the operation to work.

Reply:
Yes, that sounds like a good idea :+1: